src/hotspot/share/classfile/placeholders.cpp

```diff
@@ -116,11 +256,30 @@ Symbol* PlaceholderTable::find_entry(unsigned int hash,
   return (probe != NULL ? probe->klassname() : NULL);
 }
 
-  // find_and_add returns probe pointer - old or new
-  // If no entry exists, add a placeholder entry
-  // If entry exists, reuse entry
-  // For both, push SeenThread for classloadAction
-  // If LOAD_SUPER, this is used for circularity detection for instanceklass loading.
+static const char* action_to_string(PlaceholderTable::classloadAction action) {
+  switch (action) {
```

Comment:
Method log is the only method that calls action_to_string.  Would you consider just adding the switch statement to the log method and doing away with action_to_string?  I'm fine either way just a preference comment.

Reply:
I had a look at importing the switch on action and it's neater the way I have it so I only have one print statement like:
    ls.print("%s %s ", function, action_to_string(action));